config/identifiable.go

```diff
@@ -69,6 +69,11 @@ func (id ComponentID) Name() string {
 	return id.nameVal
 }
 
+// MarshalText implements the encoding.TextMarshaler interface.
+func (id ComponentID) MarshalText() (text []byte, err error) {
+	return []byte(id.String()), nil
+}
+
```

Comment:
Is this needed?

Reply:
This is needed because otherwise the `ComponentID` will encode as a struct, which will break the marshalling.

Reply:
you are right :) Can you add a comment that this marshals the 2 values as one string bla bla bla :))